cmake/extensions.cmake

```diff
@@ -235,9 +235,16 @@ Please provide one of following: CONF_FILES")
                         BUILD ${ZEPHYR_FILE_BUILD}
     )
     list(APPEND filename_list ${filename})
+
+    zephyr_build_string(filename
+                        BUILD ${ZEPHYR_FILE_BUILD}
+    )
+    list(APPEND filename_list ${filename})
     list(REMOVE_DUPLICATES filename_list)
 
     foreach(filename ${filename_list})
+      string(REGEX REPLACE "^_" "" filename "${filename}")
```

Comment:
Should we replace it in every filename or just in the last one?

Reply:
Does it matter? It is to get rid of leading double underscore. Making it done only for the last filename would complicate logic and make everything bug prone in case filename_list will be changed in the future.

Alternative would be to fix it while adding the name. Maybe @tejlmand has preference.

Reply:
A small hack, if you do:
```
   zephyr_build_string(filename
                        BOARD ${ZEPHYR_FILE_BUILD}
    )
    list(APPEND filename_list ${filename})
```
instead.
Then you probably get what you want and you can remove the `REGEX`.

Upstream we should probably allow `BOARD` to be an optional argument as well.

In general we should avoid generating strings that needs additional post-processing as they tend to give unpleasent surprises later.
In this case if there really was a reason to have a double `__` in the name.